### DISCUSSION TITLE
Building with Bazel

### DISCUSSION THREAD

**Discussion by Heeten** [DioxusLabs/dioxus#1340] [General]
Has anyone built Dioxus with Bazel? If not, any advice on how to get started with this for the web/wasm dioxus build?

Here's my impression so far. Looking at the comment at [cli/src/builder.rs#L25](https://github.com/DioxusLabs/dioxus/blob/master/packages/cli/src/builder.rs#L25) it looks like `dx` just uses cargo to build the code targetting wasm32, so my projects `Cargo.toml` file will have all the info I need to convert the build from cargo to bazel.

After that, the additional thing I need to do is replace `dx serve` by having something serve an index,html and wasm. In `dist` I also see `dioxus-web` and `dioxus-interpreter` in `assets/dioxus/snippets`, where do those come from, do they depend on my project/build or are they static assets that can be pulled down and served?

**Comment by ealmloff** [comment]
The javascript snippets get created by the [wasm-bindgen-cli-support crate](https://docs.rs/wasm-bindgen-cli-support/0.2.8/wasm_bindgen_cli_support/). They pull out JS from the source code of all dependancies.

**Reply by Heeten** [reply]
Ah thank you, I think that gives me a good handle on what `dx serve` does and how to manually reproduce it.
